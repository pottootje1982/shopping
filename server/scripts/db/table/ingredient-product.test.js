describe('storeMapping()', () => {
  let ingToProduct, recipeDb

  beforeAll(async () => {
    ;({ recipeDb, ingToProduct, translationsDb } = global)
  })

  it('retrieves stored translations', async () => {
    await ingToProduct.storeMapping(
      'Prei',
      { id: 1273124, title: 'prei' },
      'ah'
    )
    const mapping = await ingToProduct.getMapping('pRei', 'ah')
    expect(mapping.product).toEqual({
      id: 1273124,
      ignore: null,
      notAvailable: null,
      title: 'prei'
    })
  })

  it('updates stored translations', async () => {
    await ingToProduct.storeMapping(
      'prei',
      { id: 1273124, title: 'prei' },
      'ah'
    )
    const mapping = await ingToProduct.getMapping('prei', 'ah')
    expect(mapping.product).toEqual({
      id: 1273124,
      ignore: null,
      notAvailable: null,
      title: 'prei'
    })

    await ingToProduct.storeMapping('prei', { id: 4, title: 'AH prei' }, 'ah')
    const newMapping = await ingToProduct.getMapping('prei', 'ah')
    expect(newMapping.product).toEqual({
      id: 4,
      ignore: null,
      notAvailable: null,
      title: 'AH prei'
    })
  })

  it('retrieves stored translations for recipe', async () => {
    const recipe = await recipeDb.getRecipe(
      '3fe04f98-8d73-4e9d-a7da-f4c1241aa3c4'
    )
    await ingToProduct.storeMapping('Prei', { id: 1273124 }, 'ah')
    await ingToProduct.storeMapping('zalm', { id: 1 }, 'ah')
    await ingToProduct.storeMapping('dille', { id: 2 }, 'ah')
    await ingToProduct.storeMapping('aardappels', { id: 3 }, 'ah')

    const recipes = await ingToProduct.getMappings([recipe], 'ah')
    recipes.forEach((r) => delete r._id)
    expect(recipes).toMatchSnapshot()
  })
})
