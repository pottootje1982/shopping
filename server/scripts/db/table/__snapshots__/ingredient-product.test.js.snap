// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storeMapping() retrieves stored translations for recipe 1`] = `
Array [
  Object {
    "categories": Array [
      "cef31e04-4245-4483-9e83-34c26ebf267a",
    ],
    "cook_time": "",
    "created": "2015-08-29 11:30:47",
    "description": "",
    "difficulty": "",
    "directions": "",
    "hash": "DA475055E02DCFCBA26744E139067B6B8178D93FDE143B755FA0A8B7571610A9",
    "image_url": null,
    "in_trash": false,
    "ingredients": "Prei
Dille
Blik tomaten
Zalm
3 Aardappels
Wijn
Honing
koriander poeder",
    "is_pinned": false,
    "name": "Zalm met prei",
    "notes": "",
    "nutritional_info": "",
    "on_favorites": false,
    "on_grocery_list": null,
    "parsedIngredients": Ingredients [
      Object {
        "all": Array [
          undefined,
          undefined,
          "Prei",
        ],
        "full": "Prei",
        "ingredient": "Prei",
        "product": Object {
          "id": 1273124,
          "ignore": undefined,
          "notAvailable": undefined,
          "title": undefined,
        },
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          undefined,
          undefined,
          "Dille",
        ],
        "full": "Dille",
        "ingredient": "Dille",
        "product": Object {
          "id": 2,
          "ignore": undefined,
          "notAvailable": undefined,
          "title": undefined,
        },
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          undefined,
          "Blik",
          "tomaten",
        ],
        "full": "Blik tomaten",
        "ingredient": "tomaten",
        "product": Object {
          "id": 395307,
          "title": "AH Biologisch Tomatenblokjes",
        },
        "quantity": 1,
        "unit": "Blik",
      },
      Object {
        "all": Array [
          undefined,
          undefined,
          "Zalm",
        ],
        "full": "Zalm",
        "ingredient": "Zalm",
        "product": Object {
          "id": 1,
          "ignore": undefined,
          "notAvailable": undefined,
          "title": undefined,
        },
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          3,
          undefined,
          "Aardappels",
        ],
        "full": "3 Aardappels",
        "ingredient": "Aardappels",
        "product": Object {
          "id": 3,
          "ignore": undefined,
          "notAvailable": undefined,
          "title": undefined,
        },
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          undefined,
          undefined,
          "Wijn",
        ],
        "full": "Wijn",
        "ingredient": "Wijn",
        "product": undefined,
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          undefined,
          undefined,
          "Honing",
        ],
        "full": "Honing",
        "ingredient": "Honing",
        "product": undefined,
        "quantity": 1,
        "unit": undefined,
      },
      Object {
        "all": Array [
          undefined,
          undefined,
          "koriander poeder",
        ],
        "full": "koriander poeder",
        "ingredient": "koriander poeder",
        "product": undefined,
        "quantity": 1,
        "unit": undefined,
      },
    ],
    "photo": null,
    "photo_hash": null,
    "photo_large": null,
    "photo_url": null,
    "prep_time": "",
    "rating": 0,
    "scale": null,
    "servings": "",
    "source": "",
    "source_url": "",
    "total_time": "",
    "uid": "3fe04f98-8d73-4e9d-a7da-f4c1241aa3c4",
  },
]
`;
